### DISCUSSION TITLE
[ZK Stack] - <Failed to execute `zk stack prover-setup`>

### DISCUSSION THREAD

**Discussion by asdv23** [zkSync-Community-Hub/zksync-developers#481] [ZK Stack] [answered]
### Module Affected

- [ ] System contracts (era-system-contracts)
- [ ] Smart contracts (era-contracts)
- [ ] zkEVM (era-zk_evm)
- [ ] Server (zksync-era)
- [X] Prover (era-boojum)
- [ ] Fee withdrawer (era-fee-withdrawer)
- [ ] Block Explorer (clock-explorer)
- [ ] Portal - Wallet + Bridge (dapp-portal)
- [ ] Other (please specify below)

### Rust Version

rustc 1.74.0-nightly (5c6a7e71c 2023-08-20)

### Operating System

- [ ] Ubuntu
- [X] MacOS
- [ ] Windows
- [ ] Other (Please specify below)

### Issue Description

When I execute to `zk stack prover-setup ` it fails according to this [documentation](https://docs.zksync.io/zk-stack/running-a-hyperchain/locally.html#enabling-boojum-prover).

1. When I select `Download default keys`, it will `failed, reason: read ETIMEDOUT`
2. When I select `Generate locally`, it will called `Result::unwrap()` on an `Err` value: basic_1

Caused by:
    0: save_setup_data()
    1: Failed saving setup-data at path: "/usr/src/setup-data/setup_basic_1_data.bin"
    2: No such file or directory (os error 2)

![image](https://github.com/zkSync-Community-Hub/zksync-developers/assets/115535473/822de5bd-51bb-45bf-8b0b-71b049a7df34)
![image](https://github.com/zkSync-Community-Hub/zksync-developers/assets/115535473/5a7e13ce-7a24-4529-bb62-48d8d41fb64c)

### Expected Behavior

I can run the zk prover in the end

### Current Behavior

End of abnormal interrupt

### Repository Link (if applicable)

https://github.com/matter-labs/zksync-era/releases/tag/core-v23.1.0

### Additional Details

[zk_stack_prover-setup.log](https://github.com/zkSync-Community-Hub/zksync-developers/files/15091107/zk_stack_prover-setup.log)

### Prior Research

- [X] Yes, I searched for similar issues and did not find any matching ones.

**Comment by asdv23** [comment]
It doesn't look like there is a problem with the downloaded key files.

```bash
% ls -l -h
total 16646200
-rw-r--r--@ 1 ivan  staff    64M  4 23 18:05 setup_2^20.key
-rw-r--r--@ 1 ivan  staff   128M  4 23 18:11 setup_2^21.key
-rw-r--r--@ 1 ivan  staff   256M  4 23 18:22 setup_2^22.key
-rw-r--r--@ 1 ivan  staff   512M  4 24 10:06 setup_2^23.key
-rw-r--r--@ 1 ivan  staff   1.0G  4 24 10:16 setup_2^24.key
-rw-r--r--@ 1 ivan  staff   2.0G  4 24 10:27 setup_2^25.key
-rw-r--r--@ 1 ivan  staff   4.0G  4 22 13:37 setup_2^26.key
```

**Comment by asdv23** [comment, marked as answer]
Manually retrieve https://storage.googleapis.com/storage/v1/b/matterlabs-setup-data-us/o and found that there is a problem with the path configured by `prover/setup-data-cpu-keys.json`, and the `8ef5506` path does not exist. file, only these 4 prefixes are feasible for CPU: 
- c1c96d5 
- e01b13f 
- e2e94ff 
- f6a69e0

Manually change this json file to c1c96d5 and try again, the download is normal.
![image](https://github.com/zkSync-Community-Hub/zksync-developers/assets/115535473/2c30f299-b87f-40a5-9843-20a2e4564495)

**Reply by bxpana** [reply]
So all is good now?

**Reply by asdv23** [reply]
I'm not sure now because it's still being downloaded. So am I right to choose `c1c96d5`? Or are the other `e01b13f | e2e94ff | f6a69e0` actually the same?

**Reply by asdv23** [reply]
zk stack prover-setup is fine, but how do I run `zk prover`? https://github.com/zkSync-Community-Hub/zksync-developers/discussions/476

**Reply by hatemosphere** [reply]
@asdv23 are you using main branch? you cannot use random keys with random commit sha, just always use main, so it will be consistent and keys will be there.

but please be aware that prover stack will only work on GCP and also `zk stack init` will most likely not work at all with latest main - we are now completely rewriting it
